teuthology/misc.py

```diff
@@ -51,7 +51,13 @@ def host_shortname(hostname):
     else:
         return hostname.split('.', 1)[0]
 
-def canonicalize_hostname(hostname, user: Optional[str] ='ubuntu'):
+# sentinel value to indicate the default user value is to be used.
+# None and the empty string are reserved for specifying no user value
+# to be backwards compatible with older code.
+_default_user = '.'
+
+def canonicalize_hostname(hostname, user: Optional[str] = _default_user):
```

Comment:
Because the signature is touched, may we have consistent typing style for all parameters and function return value:
```
def canonicalize_hostname(hostname: str, user: Optional[str] = _sentinel_user) -> str:
```

Reply:
Fair enough, done.